app/feedme/ai_extraction_engine.py

```
@@ -0,0 +1,530 @@
+        text_elements = soup.find_all(text=True)
+        
+        qa_pairs = []
+        current_question = None
+        
+        for text in text_elements:
+            text = text.strip()
+            if len(text) < 20:  # Skip short text
+                continue
+            
+            # Check if it looks like a question
+            if any(pattern[0].search(text) for pattern in self.qa_patterns if pattern[1] == 'question'):
```

Pylint (3.3.7)

[error] Instance of 'str' has no 'search' member

(E1101)